lib/widget/roles_selection.dart

```diff
@@ -0,0 +1,211 @@
+          child: Text(
+            AppLocalizations.of(context).submit,
+            style: buttonTextStyle,
+          ),
+          onPressed: (_getNumRolesInTeam(currentRoleIds, Team.FRIENDLY) ==
+                      numFriendlyRoles &&
+                  _getNumRolesInTeam(currentRoleIds, Team.SCARY) ==
+                      numScaryRoles)
+              ? () => store.dispatch(new SubmitRolesAction())
+              : null,
+        ),
+      ]),
+    ));
+  }
+
+  int _getNumRolesInTeam(final Set<String> roleIds, final Team team) {
+    return roleIds
+        .where((roleId) => Roles.getTeam(roleId) == team)
+        .toList()
+        .length;
+  }
+
+  List<Widget> _getRolesList(
+      final BuildContext context,
+      final Store<GameModel> store,
+      final Team team,
+      final Color color,
+      final Set<String> selectedRoleIds,
+      final int numRolesInTeam,
+      final bool amOwner) {
+    final List<Role> teamRoles = Roles.allRoles
+        .where((role) => Roles.getTeam(role.roleId) == team)
+        .toList();
+    final List<Widget> rolesList = [];
+    rolesList.addAll(new List.generate(teamRoles.length, (i) {
+      final Role role = teamRoles.elementAt(i);
+      final String roleId = role.roleId;
+      final bool compulsory = role.isCompulsory();
+      final Text text = Text(Roles.getRoleDisplayName(context, roleId),
+          style: buttonTextStyle, textAlign: TextAlign.center);
+      final Widget content = compulsory
+          ? iconText(new Icon(Icons.star, color: Colors.white), text)
+          : text;
+      return new Padding(
+          padding: paddingNano,
+          child: new RaisedButton(
+              color: selectedRoleIds.contains(roleId) ? color : Colors.grey,
+              child: content,
+              onPressed: () => amOwner
+                  ? selectedRoleIds.contains(roleId)
+                      ? compulsory
+                          ? _showCantDoActionDialog(
+                              context,
+                              AppLocalizations.of(context).compulsoryRoles(
+                                  Roles.getRoleDisplayName(
+                                      context, Roles.bertie.roleId),
+                                  Roles.getRoleDisplayName(
+                                      context, Roles.brenda.roleId)))
+                          : store.dispatch(new RemoveRoleAction(roleId))
+                      : _getNumRolesInTeam(selectedRoleIds, team) ==
+                              numRolesInTeam
+                          ? _showCantDoActionDialog(
+                              context, AppLocalizations.of(context).fullTeam)
+                          : store.dispatch(new AddRoleAction(roleId))
+                  : _showCantDoActionDialog(context,
+                      AppLocalizations.of(context).onlyOwnerModifies)));
+    }));
+    return rolesList;
+  }
+
+  _showCantDoActionDialog(final BuildContext context, final String content) {
```

Comment:
I reckon that we should have a text area which shows a description of whatever role you last clicked on. This space could also show these `can't do` errors, and might be less disruptive than a pop-up.

Reply:
I can do it with more time.
However, I'm not sure this is the best way to display that info. I think it's confusing to show something both when roles have been selected and deselected and I think there should be a way of learning about roles that doesn't imply deselecting something else to select the role you want to know about.